### DISCUSSION TITLE
[Auth0 client] Redis authentication

### DISCUSSION THREAD

**Discussion by georgev-97** [mia-platform/community#364] [Q&A] [answered]
### Which section?

Console

### Which is the version?

12

### Description

Hello guys,
the Auth0 client (according to the [documentation](https://docs.mia-platform.eu/docs/runtime_suite/auth0-client/configuration)) accepts only the env REDIS_HOSTS to configure Redis, what is the correct format for this variable to include username and password?
I've tried adding REDIS_USERNAME and REDIS_PASSWORD as in the authentication-service but it doesn't seem to work since I get the error: `NOAUTH Authentication required`

**Comment by malta895** [comment, marked as answer]
Hi @georgev-97,
unfortunately at the moment we do not support authenticated Redis connections with the auth0-client: you can only use a Redis instance without password protection.

We have planned the implementation of the Redis authentication support, we will keep you posted about it!

Thank you for your feedback!

**Reply by georgev-97** [reply]
Ok, thank you

**Comment by fredmaggiowski** [comment]
Hi @georgev-97 I've just submitted a PR to the service in order to support username and password for Redis authentication; it was pretty straightforward; as soon as it passes our internal QA I'll bump a new version and let you know :)

**Reply by fredmaggiowski** [reply]
You can use v3.3.2 and set the env `REDIS_USERNAME` and `REDIS_PASSWORD` (as you already tried)

Please note that username and password are not yet supported if you are using sentinel mode

**Reply by georgev-97** [reply]
Very nice, thank you!!

**Reply by georgev-97** [reply]
The authentication on redis now works just fine, but when I try to call /oauth/token the auth0 client gives me another redis related error:` {"error":"key not found into redis","level":50,"msg":"state not present in redis","reqId":"ce5e3cf9-6bc0-402f-9c69-971728385c5b","time":1704896676812}` do you know why?

**Reply by fredmaggiowski** [reply]
You are welcome, let us know if you find any issue 🥳

**Reply by fredmaggiowski** [reply]
>but when I try to call /oauth/token the auth0 client gives me another redis related error: {"error":"key not found into redis","level":50,"msg":"state not present in redis","reqId":"ce5e3cf9-6bc0-402f-9c69-971728385c5b","time":1704896676812} do you know why?

This seems like a problem with OAuth2 flow (e.g. not using the `/authorize` endpoint before invoking for the `/oauth/token`), I'd suggest you to open a different discussion with all the related information so that we can address this properly :)
